Assets/Feature01/Scripts/CharacterStates/PolarityManager.cs

```diff
@@ -0,0 +1,84 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PolarityManager : MonoBehaviour
+{
+
+
+    public float MaxHealth;
+    public float CurrentHealth;
+    public float MovementSpeed;
+    public Color PlayerColor;
+    public Stats CurrentPlayerStats;
+    public PlayerState CurrentPlayerState;
+
+    [SerializeField] ShootSystem ShootSystem;
+    [SerializeField] PlayerController PlayerController;
+
+
+    Dictionary<PlayerState, Stats> PlayerStatesSelection = new Dictionary<PlayerState, Stats>();
+    
+   
+
+
+    private void Awake()
+    {
+        PlayerStatesSelection =  new Dictionary<PlayerState, Stats>()
```

Comment:
This is okay. But you should probably drag drop them from the inspector instead of using resources folder like this, Then again this could be fine in the long run.